src/games/strategy/engine/data/Route.java

```diff
@@ -355,8 +355,8 @@ public String toString() {
    *         the route consists of only a starting territory
    */
   public Territory getEnd() {
-    if (m_steps.size() == 0) {
-      return null;
+    if (m_steps.isEmpty()) {
+      return m_start;
```

Comment:
Alright. I'm ok moving forward with the change. I'd prefer changes like this in separate PRs or if in a PR like this then definitely commented since its pretty unclear why the change is being made.